schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java

```diff
@@ -538,6 +529,16 @@ public SchemaIdVersion mergeSchemaVersion(Long schemaVersionId, SchemaVersionMer
         return schemaVersionLifecycleManager.mergeSchemaVersion(schemaVersionId, schemaVersionMergeStrategy);
     }
 
+    @Override
+    public Collection<AggregatedSchemaBranch> getAggregatedSchemaBranch(String schemaName) throws SchemaNotFoundException, SchemaBranchNotFoundException {
+        Collection<AggregatedSchemaBranch> aggregatedSchemaBranches = new LinkedList<>();
```

Comment:
nit: may want to use arraylist to avoid creating more references with linkedlist, there may not be  large no of branches for a schema.